vtex-apps/slider-layout
# Fix click handlers on PaginationDots component

#### What does this PR do?

The handleDotClick method used by PaginationDots component was trying to mutate the `SliderState` directly, using a dispatch call. However, the value for the state variable `currentSlide` was wrongly calculated, which meant the SliderState was not updated as expected.

This incongruence caused slides to not be rendered when users clicked in a PaginationDot to navigate since the `useSliderVisibility` hook would be fooled into returning `false` for the `isItemVisible` variable, even though slides would be visible.

#### How to test it?

1. To see the current behavior, go to https://storetheme.vtex.com and try to navigate to the second page in the Shelf by clicking in the second navigation dot. You should see that some elements were not rendered;
2. Go back to the first page and navigate to the next using the forward arrow in the right. Now everything should be working correctly;
3. To test this fix, go to this [Workspace](https://victorhmp--storecomponents.myvtex.com/) and replicate step 1. Now the elements should be rendering correctly :) .

#### Related to / Depends on \*

This closes https://github.com/vtex-apps/store-discussion/issues/291
